kernel/src/snapshot.rs

```diff
@@ -71,6 +72,26 @@ impl Snapshot {
         Self::try_new_from_log_segment(table_root, log_segment, engine)
     }
 
+    /// Create a new [`Snapshot`] instance from an existing [`Snapshot`]. This is useful when you
+    /// already have a [`Snapshot`] lying around and want to do the minimal work to 'update' the
+    /// snapshot to a later version.
```

Comment:
Just to clarify, is this api only for versions later than the existing snapshot?